pkg/cache/informer_cache.go

```diff
@@ -145,13 +145,32 @@ func (ic *informerCache) GetInformer(ctx context.Context, obj client.Object) (In
 		return nil, err
 	}
 
-	_, i, err := ic.Informers.Get(ctx, gvk, obj)
+	_, i, err := ic.getInformerForKind(ctx, gvk, obj)
 	if err != nil {
 		return nil, err
 	}
 	return i.Informer, err
 }
 
+func (ic *informerCache) getInformerForKind(ctx context.Context, gvk schema.GroupVersionKind, obj runtime.Object) (bool, *internal.Cache, error) {
+	var cache *internal.Cache
+	var started bool
+	var err error
+	switch ic.missPolicy {
+	case CacheMissPolicyFail:
+		var ok bool
+		cache, started, ok = ic.Informers.Peek(gvk, obj)
+		if !ok {
+			err = &ErrResourceNotCached{gvk: gvk}
+		}
+	case CacheMissPolicyBackfill:
+		fallthrough
+	default:
+		started, cache, err = ic.Informers.Get(ctx, gvk, obj)
```

Comment:
Shouldn't the `Get` go into backfill, and we error out in any other case?

Reply:
Go's type system is not expressive enough to capture that there are only two valid options for the miss policy here. I wanted to write this in a way such that the reviewers would be more confident that the default behavior did not change from the previous behavior. No strong opinions here but not having a valid miss policy here is a programmer error / panic and should not be possible at runtime.

Reply:
Let's not have two options that do the same thing, if we have a default == empty which is to backfill we can try to express that through an empty string option maybe, or just offer `Fail` for now as a non-empty option?

Reply:
There are no two options that do the same thing. You just can't express that the type has two valid enum cases, and not the empty string.

Reply:
> the value of the miss policy may only ever be CacheMissPolicyBackfill or CacheMissPolicyFail.

For this reason, I don't see what's wrong with a `panic` in the case that the value is not one of those two options.

Reply:
I mean, what's the _value_ in a panic?